packages/core/ui-settings/core-ui-settings-server-internal/src/create_or_upgrade_saved_config/get_upgradeable_config.ts

```diff
@@ -47,11 +51,26 @@ export async function getUpgradeableConfig({
     });
 
   // try to find a config that we can upgrade
-  const findResult = savedConfigs.find((savedConfig) =>
+  const matchingResults = savedConfigs.filter((savedConfig) =>
     isConfigVersionUpgradeable(savedConfig.id, version)
   );
-  if (findResult) {
-    return { id: findResult.id, attributes: findResult.attributes };
+  const mostRecentConfig = getMostRecentConfig(matchingResults);
+  if (mostRecentConfig) {
+    return { id: mostRecentConfig.id, attributes: mostRecentConfig.attributes };
   }
```

Comment:
The root change. The code was making the assumption that sorting by `buildNum` was sufficient, but this field is indexed as `keyword`, not `number`, which cause the order to be wrong in some situations.

E.g

```ts
["9517", "16602", "63240"].sort()
>> (3) ['16602', '63240', '9517']
```

So we're now manually comparing the version of each matching document to make sure that the most recent is used.